web-platform/website/src/services/UserDocumentService.tsx

```diff
@@ -137,6 +137,11 @@ async function updateUserDocument(
 ): Promise<void> {
   assert(partialDocument, "Invalid user document.");
 
+  if (!firebaseUid) {
+    console.debug("Not signed in yet, user document cannot be modified.");
+    return;
+  }
+
```

Comment:
We should throw here IMO so we can catch and place checks.

Reply:
assert(firebaseUid, "Attempt to update user document when no user is logged in.");